### PR TITLE
Feature/uppsf 1629 change notification subscriptions

### PR DESCRIPTION
**Pull request by gotha**
# Description

## What

Allow notifications for LiveBlogPost and LiveBlogPackage when client specifies them as content types. 
Allow multiple types subscriptions. 

## Why

https://financialtimes.atlassian.net/browse/UPPSF-1629

## Anything, in particular, you'd like to highlight to reviewers

none

## Scope and particulars of this PR (Please tick all that apply)

- [ ] Tech hygiene (dependency updating & other tech debt)
- [ ] Bug fix
- [x] Feature
- [ ] Documentation
- [ ] Breaking change
- [ ] Minor change (e.g. fixing a typo, adding config)

___
This Pull Request follows the rules described in our [Pull Requests Guide](https://github.com/Financial-Times/upp-docs/tree/master/guides/pr-guide)

### CODE REVIEW

File: dispatch/dispatcher.go [hunk 1]
```diff
@@ -75,12 +75,12 @@ func (d *Dispatcher) Subscribers() []Subscriber {
 	return subs
 }
 
-func (d *Dispatcher) Subscribe(address string, subType string, monitoring bool) Subscriber {
+func (d *Dispatcher) Subscribe(address string, subTypes []string, monitoring bool) Subscriber {
```
Comment (ivan-p-nikolov) [inline dispatch/dispatcher.go:78]: We should probably add a test for multiple subscription types. Just to make clear what's the expected behavior of this functionality.

File: resources/push.go [hunk 2]
```diff
@@ -171,17 +172,29 @@ func getClientAddr(r *http.Request) string {
...
+	if len(subTypes) == 0 {
+		return []string{defaultSubscriptionType}, nil
 	}
-	for _, t := range supportedSubscriptionTypes {
-		if strings.EqualFold(subType, t) {
-			return subType, nil
+	// subTypes are being send by the client (subscriber), and needs to be matched with such string value
+	for _, subType := range subTypes {
+		if subType == dispatch.AllContentType {
+			retVal = append(retVal, dispatch.AllAllowedList...)
```
Comment (ivan-p-nikolov) [inline resources/push.go:185]: Why keep the `AllAllowedList` as a global in `dispatch` when we only use it here?
I would make the list a member of the SubHandler and pass it as a parameter to resolveSubType. I don't trust global variables.

**Review by rnov — APPROVED** [review]

File: helm/notifications-push/app-configs/notifications-push_eks_delivery.yaml [hunk 3]
```diff
@@ -9,6 +9,7 @@ env:
   CONSUMER_BACKOFF: "2"
   NOTIFICATIONS_RESOURCE: "content"
   CONTENT_URI_WHITELIST: "^http://(methode|wordpress-article|content)(-collection|-content-placeholder)?-(mapper|unfolder)(-pr|-iw)?(-uk-.*)?\\.svc\\.ft\\.com(:\\d{2,5})?/(content|complementarycontent)/[\\w-]+.*$"
-  CONTENT_TYPE_WHITELIST: "application/vnd.ft-upp-article+json,application/vnd.ft-upp-content-package+json,application/vnd.ft-upp-audio+json"
+  CONTENT_TYPE_WHITELIST: "application/vnd.ft-upp-article+json,application/vnd.ft-upp-content-package+json,application/vnd.ft-upp-audio+json,application/vnd.ft-upp-live-blog-post+json,application/vnd.ft-upp-live-blog-package+json"
   WHITELISTED_METADATA_ORIGIN_SYSTEM_HEADERS: "http://cmdb.ft.com/systems/pac, http://cmdb.ft.com/systems/methode-web-pub, http://cmdb.ft.com/systems/next-video-editor"
+  ALLOWED_ALL_CONTENT_TYPE: "Article,ContentPackage,Audio"
```
Comment (ivan-p-nikolov) [inline helm/notifications-push/app-configs/notifications-push_eks_delivery.yaml:14]: We probably need to add `ContentPlaceholder` in the helm configs.

Comment (gotha) [reply helm/notifications-push/app-configs/notifications-push_eks_delivery.yaml:14]: you're absolutely right

**Review by ivan-p-nikolov — APPROVED** [review]

**Pull request merged**